src/support.js

```diff
@@ -612,21 +613,22 @@ var functionPointers = new Array(jsCallNumSigs * {{{ RESERVED_FUNCTION_POINTERS
 var jsCallStartIndex = 1;
 var functionPointers = new Array({{{ RESERVED_FUNCTION_POINTERS }}});
 #endif // WASM_BACKEND && RESERVED_FUNCTION_POINTERS
+#endif // EMULATED_FUNCTION_POINTERS == 0
 
-// 'sig' parameter is only used on LLVM wasm backend
+// 'sig' parameter is currently only used for LLVM backend under certain
+// circumstance: RESERVED_FUNCTION_POINTERS=1, EMULATED_FUNCTION_POINTERS=0.
 function addFunction(func, sig) {
-#if WASM_BACKEND
-  assert(typeof sig !== 'undefined',
-         'Second argument of addFunction should be a wasm function signature ' +
-         'string');
-#endif // WASM_BACKEND
-#if ASSERTIONS
+#if ASSERTIONS == 2
   if (typeof sig === 'undefined') {
-    err('warning: addFunction(): You should provide a wasm function signature string as a second argument. This is not necessary for asm.js and asm2wasm, but is required for the LLVM wasm backend, so it is recommended for full portability.');
+    err('warning: addFunction(): You should provide a wasm function signature string as a second argument. This is not necessary for asm.js and asm2wasm, but can be required for the LLVM wasm backend, so it is recommended for full portability.');
```

Comment:
Does this mean, now, when `RESERVED_FUNCTION_POINTERS == 1`, do we get a warning and then an error with the same message? Seems weird, but if we are gonna remove this option entirely maybe not a big deal though.

Reply:
Yes, I'm not to worried about this. 

Its important to me that the assert doesn't fire in cases where `sig` is not needed which is why I moved it down.  But i wanted to preserve the warning in all cases so as to continue to warn people against possilbe future breakage.

Reply:
I see, but I guess if you remove the `RESERVED_FUNCTION_POINTER` option for wasm backend entirely, we may not need to second argument at all by then.